.dotnet/tests/Files/FileTests.Uploads.cs

```diff
@@ -0,0 +1,242 @@
+        Assert.That(uploadDetails.Purpose, Is.EqualTo("fine-tune"));
+        Assert.That(uploadDetails.Status, Is.EqualTo("pending"));
+        Assert.That(uploadDetails.ExpiresAt, Is.GreaterThan(uploadDetails.CreatedAt));
+    }
+
+    [Test]
+    public async Task AddUploadPartWorks()
+    {
+        FileClient fileClient = GetTestClient();
+        UploadDetails uploadDetails = await CreateTestUploadAsync(fileClient);
+        using MultipartFormDataBinaryContent content = new();
+
+        content.Add([1, 2, 3, 4], "data", "data", "application/octet-stream");
+
+        ClientResult result = IsAsync
+            ? await fileClient.AddUploadPartAsync(uploadDetails.Id, content, content.ContentType)
+            : fileClient.AddUploadPart(uploadDetails.Id, content, content.ContentType);
+
+        BinaryData response = result.GetRawResponse().Content;
+        using JsonDocument jsonDocument = JsonDocument.Parse(response);
+        UploadPartDetails uploadPartDetails = GetUploadPartDetails(jsonDocument);
+
+        Assert.That(uploadPartDetails.Id, Does.StartWith("part_"));
+        Assert.That(uploadPartDetails.Object, Is.EqualTo("upload.part"));
+        Assert.That(uploadPartDetails.CreatedAt, Is.GreaterThanOrEqualTo(uploadDetails.CreatedAt));
+        Assert.That(uploadPartDetails.UploadId, Is.EqualTo(uploadDetails.Id));
+    }
+
+    [Test]
+    public async Task CompleteUploadWorks()
+    {
+        FileClient fileClient = GetTestClient();
+        UploadDetails createdUploadDetails = await CreateTestUploadAsync(fileClient);
+        using MultipartFormDataBinaryContent firstPartContent = new();
+        using MultipartFormDataBinaryContent secondPartContent = new();
+
+        firstPartContent.Add([1, 2, 3, 4], "data", "data", "application/octet-stream");
+        secondPartContent.Add([5, 6, 7, 8], "data", "data", "application/octet-stream");
+
+        UploadPartDetails firstPartDetails = await AddTestUploadPartAsync(fileClient, createdUploadDetails.Id, firstPartContent);
+        UploadPartDetails secondPartDetails = await AddTestUploadPartAsync(fileClient, createdUploadDetails.Id, secondPartContent);
+
+        BinaryContent content = BinaryContent.Create(BinaryData.FromObjectAsJson(new
+        {
+            part_ids = new[] {
+                firstPartDetails.Id,
+                secondPartDetails.Id
+            }
+        }));
+
+        ClientResult result = IsAsync
+            ? await fileClient.CompleteUploadAsync(createdUploadDetails.Id, content)
+            : fileClient.CompleteUpload(createdUploadDetails.Id, content);
+
+        BinaryData response = result.GetRawResponse().Content;
+        using JsonDocument jsonDocument = JsonDocument.Parse(response);
+        JsonElement fileElement = jsonDocument.RootElement.GetProperty("file");
+        string fileId = fileElement.GetProperty("id").GetString();
+
+        await fileClient.DeleteFileAsync(fileId);
```

Comment:
Before deleting the file, would it be possible to use the FileClient's DownloadFile method to validate that the content is equal to the first part and second part in the right order?

Reply:
I gave it a try, but I'm facing some issues when downloading the file.
- If I set the purpose to "vision", upload creation fails with a 500 error (not on us).
- If I set the purpose to anything else, we get the following error when trying to download the file:
```
Not allowed to download files of purpose: <purpose>
```

I'm not sure if that's a bug or a limitation of our subscription, but it doesn't seem to be our fault.

Reply:
Ah, I think this is a limitation of the API. Alright, no worries. Thank you for looking into it!